src/rules/quotemarkRule.ts

```diff
@@ -67,19 +69,20 @@ class QuoteWalker extends Lint.RuleWalker {
             const firstCharacter = text.charAt(0);
             const lastCharacter = text.charAt(text.length - 1);
 
-            if (this.quoteMark === QuoteMark.SINGLE_QUOTES) {
-                if (firstCharacter !== "'" || lastCharacter !== "'") {
-                    failure = this.createFailure(position, width, Rule.SINGLE_QUOTE_FAILURE);
-                }
-            } else if (this.quoteMark === QuoteMark.DOUBLE_QUOTES) {
-                if (firstCharacter !== "\"" || lastCharacter !== "\"") {
-                    failure = this.createFailure(position, width, Rule.DOUBLE_QUOTE_FAILURE);
+            const includesOtherQuoteMark = (this.quoteMark === QuoteMark.SINGLE_QUOTES) ? "'" : "\"";
```

Comment:
wait, I think you renamed the wrong variable here -- this one should be `expectedQuoteMark`